test/e2e/node_feature_discovery_test.go

```diff
@@ -828,15 +897,28 @@ func getNonControlPlaneNodes(cli clientset.Interface) ([]corev1.Node, error) {
 	return out, nil
 }
 
-// nfdLabels gets labels that are in the nfd label namespace.
-func nfdLabels(labels map[string]string) k8sLabels {
-	ret := k8sLabels{}
-
-	for key, val := range labels {
-		if strings.HasPrefix(key, nfdv1alpha1.FeatureLabelNs) {
-			ret[key] = val
+func getNodeLabels(nodes []corev1.Node, nodeName string) map[string]string {
+	for _, node := range nodes {
+		if node.Name == nodeName {
+			return node.Labels
 		}
 	}
-	return ret
+	return nil
+}
 
+func mergeMaps(a map[string]string, b map[string]string) map[string]string {
+	result := map[string]string{}
+	if a == nil {
+		return b
+	}
+	if b == nil {
+		return a
+	}
```

Comment:
nit: i'd suggest this to work consistently and always return a new copy, i.e. never return the original maps

```suggestion
	if a == nil {
		return b
	}
	if b == nil {
		return a
	}
```

Reply:
Nil checks are useless actually in this case, I can iterate directly on these maps, it won't cause any issue.